utils/python/CIME/hist_utils.py

```diff
@@ -0,0 +1,175 @@
+"""
+Functions for actions pertaining to history files.
+"""
+
+from CIME.XML.standard_module_setup import *
+
+import logging, glob, os, shutil, re
+
+def _iter_model_file_substrs(case):
+
+    models = case.get_compset_components()
+    models.append('cpl')
+    for model in models:
+        yield model
+
+def _get_all_hist_files(testcase, model, suffix="", from_dir=os.getcwd()):
+    suffix = (".%s" % suffix) if suffix else ""
+    test_hists = glob.glob("%s/%s.%s*.h?.nc%s" % (from_dir, testcase, model, suffix))
+    test_hists.extend(glob.glob("%s/%s.%s*.h.nc%s" % (from_dir, testcase, model, suffix)))
+    test_hists.extend(glob.glob("%s/%s.%s*.h?.*.nc%s" % (from_dir, testcase, model, suffix)))
+    test_hists.extend(glob.glob("%s/%s.%s*.h.*.nc%s" % (from_dir, testcase, model, suffix)))
+    if suffix == "":
+        test_hists.extend(glob.glob("%s/%s.h.nc" % (from_dir, model)))
+        test_hists.extend(glob.glob("%s/%s.h?.nc" % (from_dir, model)))
+    test_hists.sort()
+    return test_hists
+
+def move(case, suffix):
```

Comment:
For this and other functions in this file - at least for public functions: Can you please add documentation? At the least, I'd like to have documentation on each of the arguments to each function.